File: src/band/room.ts
export interface Message {
  id: string;
  agentId: string;
  role: "legal" | "finance" | "risk" | "ops" | "system";
  content: string;
  timestamp: number;
}

export class BandRoom {
  private messages: Message[] = [];

  postMessage(agentId: string, role: Message["role"], content: string) {
    const message: Message = {
      id: Math.random().toString(36).substring(7),
      agentId,
      role,
      content,
      timestamp: Date.now(),
    };
    this.messages.push(message);
    return message;
  }

  getTranscript(): Message[] {
    return this.messages;
  }

  clear() {
    this.messages = [];
  }
}

File: backend/band/room.py
import time
import uuid

class BandRoom:
    def __init__(self):
        self.messages = []

    def post_message(self, agent_id: str, role: str, content: str):
        message = {
            "id": str(uuid.uuid4())[:8],
            "agentId": agent_id,
            "role": role,
            "content": content,
            "timestamp": int(time.time() * 1000)
        }
        self.messages.append(message)
        return message

    def get_transcript(self):
        return self.messages

    def clear(self):
        self.messages = []
